structure_tests/run.sh

```diff
@@ -74,4 +76,8 @@ if [ -z "$IMAGE_NAME" ]; then
 	usage
 fi
 
-docker run --privileged=true -v /workspace:/workspace --entrypoint="$ENTRYPOINT" "$IMAGE_NAME" -c "$CMD_STRING"
+# Get id of container we're currently in.  This method is
+# system-specific but probably better than looking at $HOSTNAME.
+THIS_CONTAINER=$(basename $(head -1 /proc/self/cgroup))
```

Comment:
this seems to work as far as I can tell. just curious why you think it's better than looking at hostname? also seems to work in all cases for me.